server/src/core/event/event.resolver.ts

```diff
@@ -0,0 +1,49 @@
+import { Resolver, Mutation, Args, Context } from '@nestjs/graphql';
+import { EventService } from './event.service';
+import { Event } from './event.entity';
+import { CreateEventInput } from './dto/create-event.input';
+import axios from 'axios';
+import { OptionalJwtAuthGuard } from 'src/guards/optional-jwt.auth.guard';
+import { UseGuards } from '@nestjs/common';
+import { AuthWorkspace } from 'src/decorators/auth-workspace.decorator';
+import { User, Workspace } from '@prisma/client';
+import { AuthUser } from 'src/decorators/auth-user.decorator';
+import { anonymize } from 'src/utils/anonymize';
+
+@UseGuards(OptionalJwtAuthGuard)
+@Resolver(() => Event)
+export class EventResolver {
+  constructor(private readonly eventService: EventService) {}
+
+  @Mutation(() => Event)
+  createEvent(
+    @Args() createEventInput: CreateEventInput,
+    @Context() context: any,
+    @AuthWorkspace() workspace: Workspace,
+    @AuthUser() user: User,
+  ) {
+    if (process.env.IS_TELEMETRY_ENABLED === 'false') {
+      return;
+    }
+
+    const data = {
+      type: createEventInput.type,
+      data: {
+        userUUID: user ? anonymize(user.id) : undefined,
+        workspaceUUID: workspace ? anonymize(workspace.id) : undefined,
+        workspaceDomain: workspace ? workspace.domainName : undefined,
+        ...JSON.parse(createEventInput.data),
+      },
+    };
+
+    axios
```

Comment:
@FelixMalfait This should be in the `EventService`, and you shouldn't directly use `axios` but instead use `HttpService` from `@nestjs/axios`.
(You need to add `HttpModule` and inject `HttpService` into `EventService`)

Reply:
Good point thanks!